ui/src/app/cohort-search/demographics/demographics.component.tsx

```diff
@@ -0,0 +1,592 @@
+    borderRadius: '3px',
+    fontSize: '0.5rem',
+    fontWeight: 300,
+    marginTop: '0.25rem',
+    padding: '0 0.5rem',
+    width: '1rem',
+  },
+  ageLabel: {
+    fontSize: '14px',
+    fontWeight: 600,
+    color: colors.primary
+  },
+  agePreview: {
+    minWidth: '50%',
+    padding: '0.25rem 1rem',
+    width: 'auto'
+  },
+  calculateBtn: {
+    background: colors.primary,
+    border: 'none',
+    borderRadius: '0.3rem',
+    color: colors.white,
+    cursor: 'pointer',
+    fontSize: '12px',
+    height: '1.5rem',
+    letterSpacing: '0.02rem',
+    lineHeight: '0.75rem',
+    margin: '0.25rem 0.5rem 0.25rem 0',
+    padding: '0rem 0.75rem',
+    textTransform: 'uppercase',
+  },
+  count: {
+    alignItems: 'center',
+    background: colors.accent,
+    borderRadius: '10px',
+    color: colors.white,
+    display: 'inline-flex',
+    fontSize: '10px',
+    height: '0.625rem',
+    justifyContent: 'center',
+    lineHeight: 'normal',
+    margin: '0 0.25rem',
+    minWidth: '0.675rem',
+    padding: '0 4px',
+    verticalAlign: 'middle'
+  },
+  countPreview: {
+    backgroundColor: colorWithWhiteness(colors.secondary, 0.8),
+    padding: '0.5rem',
+    margin: '0 2.5%',
+    position: 'absolute',
+    width: '95%',
+    bottom: '0.5rem',
+  },
+  option: {
+    color: colors.black,
+    cursor: 'pointer',
+    fontSize: '13px',
+    fontWeight: 400,
+    marginBottom: '0.5rem',
+    padding: '0 0.25rem',
+    textTransform: 'capitalize',
+  },
+  resultText: {
+    color: colors.primary,
+    fontWeight: 500,
+  },
+  selectIcon: {
+    color: colors.select,
+    marginRight: '0.25rem'
+  },
+  selected: {
+    cursor: 'not-allowed',
+    opacity: 0.4
+  },
+  selectList: {
+    alignItems: 'center',
+    display: 'flex',
+    marginRight: '1rem',
+    maxHeight: '15rem',
+    padding: '0.5rem 0 0 1rem'
+  },
+  slider: {
+    flex: 1,
+    padding: '0 0.5rem',
+    margin: '0 1rem',
+  },
+  sliderContainer: {
+    alignItems: 'center',
+    display: 'flex',
+    marginRight: '1rem',
+    paddingLeft: '1rem',
+    width: '96%',
+  }
+});
+// Template node used for age selections
+const ageNode = {
+  hasAncestorData: false,
+  attributes: [],
+  code: '',
+  domainId: DomainType.PERSON,
+  group: false,
+  name: 'Age',
+  parameterId: 'age-param',
+  isStandard: true,
+  type: CriteriaType.AGE,
+  value: ''
+};
+
+const ageTypes = [
+  {label: 'Current Age', type: AttrName.AGE.toString()},
+  {label: 'Age at Consent', type: AttrName.AGEATCONSENT.toString()},
+  {label: 'Age at CDR Date', type: AttrName.AGEATCDR.toString()}
+];
+
+const defaultMinAge = '18';
+const defaultMaxAge = '120';
+
+function sortByCountThenName(critA, critB) {
+  const A = critA.count || 0;
+  const B = critB.count || 0;
+  const diff = B - A;
+  return diff === 0
+        ? (critA.name > critB.name ? 1 : -1)
+        : diff;
+}
+interface Props {
+  select: Function;
+  selectedIds: Array<string>;
+  selections: Array<any>;
+  wizard: any;
```

Comment:
Can you type this as a React.Element or something? Just is there anything more specific than `any` you can use?